### PR TITLE
Version mandatory for PUT Files and Bundles

### PR DESCRIPTION
**Pull request by Bento007**
### Test plan
- Updated test cases to always include version
- Added test cases to test when no version is present

### Deployment instructions & migrations
- the HCA-CLI will need to be updated to handles these changes.

### Release notes
- updated the swagger paths for bundles/put to require version
- updated the swagger paths for files/put to require version

### CODE REVIEW

File: tests/test_bundle.py [hunk 1]
```diff
@@ -258,20 +257,31 @@ def _test_bundle_put(self, replica: Replica, fixtures_bucket: str):
             requests.codes.conflict,
         )
 
-        # should *NOT* be able to upload a bundle with a missing file, but we should get requests.codes.conflict.
+        print('should *NOT* be abke to this without bundle version.')
```
Comment (xbrianh) [inline tests/test_bundle.py]: Spelling.

This test used to be addressing `requests.codes.conflict`. Has that changed?

**Comment by Bento007** [comment]
Smoke test does pass, but I think retries in the CLI can still cause duplicate bundles.

**Comment by Bento007** [comment]
Here is the update for the CLI https://github.com/HumanCellAtlas/dcp-cli/pull/135

**Comment by Bento007** [comment]
I broke it into subtests since many of the tests are independent of each other.

**Pull request merged**